src/saturn_engine/worker/inventories/joined_sub.py

```diff
@@ -0,0 +1,94 @@
+import typing as t
+
+import abc
+import dataclasses
+import json
+from collections.abc import AsyncIterator
+
+from saturn_engine.core.api import InventoryItem
+from saturn_engine.worker.services import Services
+
+from . import Item
+from . import IteratorInventory
+
+
+class JoinedSubInventory(IteratorInventory, abc.ABC):
+    @dataclasses.dataclass
+    class Options:
+        inventory: InventoryItem
+        sub_inventory: InventoryItem
+        batch_size: int = 10
+        flatten: bool = False
+        alias: t.Optional[str] = None
+
+    def __init__(self, options: Options, services: Services, **kwargs: object) -> None:
+        super().__init__(
+            options=options,
+            services=services,
+            batch_size=options.batch_size,
+            **kwargs,
+        )
+        self.flatten: bool = options.flatten
+        self.alias: t.Optional[str] = options.alias
+        self.inventory_name = options.inventory.name
+        self.sub_inventory_name = options.sub_inventory.name
+
+        # This imports must be done late since work_factory depends on this module.
+        from saturn_engine.worker.work_factory import build_inventory
+        from saturn_engine.worker.work_factory import build_sub_inventory
+
+        self.inventory = build_inventory(options.inventory, services=services)
```

Comment:
We might have to pass kwargs? In that case, do we have to separate them between two inventories